express-delete/index.js

```diff
@@ -0,0 +1,41 @@
+const express = require('express');
+const app = express();
+var grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+
+app.get('/api/grades', (req, res) => {
+  var gradesArray = [];
+  for (var property in grades) {
+    gradesArray.push(grades[property]);
+  }
+  res.json(gradesArray);
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  delete grades[req.params.id];
+  res.sendStatus(204);
+  res.send(grades);
```

Comment:
- You can't send multiple responses to the same request (`sendStatus` sends its own response).
- A `204 No Content` response should not have a body.